packages/turbo-gen/src/utils/getWorkspaceList.ts

```diff
@@ -0,0 +1,42 @@
+import type { Project, Workspace } from "@turbo/workspaces";
+import inquirer from "inquirer";
+import {
+  getWorkspaceStructure,
+  getGroupFromWorkspace,
+} from "./getWorkspaceStructure";
+import { WorkspaceType } from "../generators/types";
+
+export function getWorkspaceList({
+  project,
+  what,
+  showAllDependencies,
+}: {
+  project: Project;
+  what: WorkspaceType;
+  showAllDependencies?: boolean;
+}): Array<Workspace | inquirer.Separator> {
+  const structure = getWorkspaceStructure({ project });
+  const workspaceChoices: Array<Workspace | inquirer.Separator> = [];
+
+  let workspacesForDisplay: Array<Workspace> = project.workspaceData.workspaces;
+  if (!showAllDependencies) {
+    if (what === "app" && structure.hasRootApps) {
+      workspacesForDisplay = structure.workspacesByGroup.apps;
+    } else if (what === "package" && structure.nonAppWorkspaces.length > 0) {
+      workspacesForDisplay = structure.nonAppWorkspaces;
+    }
+  }
+
+  // build final list with separators between groups
+  let lastGroup: string | undefined;
+  workspacesForDisplay.forEach((workspace) => {
```

Comment:
Yea this is fair, I need to get in the habit of using `for...of`. I generally reach for `forEach` when I need to create a collection from an `iterable`, but not every item of the iterable will be included. Because the alternative is `.map(...).filter(...)` which is two iterations instead of the one from the `forEach`.